### PR TITLE
Add foundation for GA4 with AMP.

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

Addresses issue:

- #7221

## Relevant technical choices

- Largely follows the approach used by the GA4 web tag

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by tofumatt — APPROVED** [review]
Code here looks good, and when I tested I saw my visits appear in the GA4 Analytics Realtime data for the property I had set up on my site when I enabled the AMP plugin. 👍🏻

**Comment by tofumatt** [comment]
(Moving to MR as @aaemnnosttv said it'd be good to have a few eyes on this, but everything looked good to me. 🙂)

File: includes/Modules/Analytics/AMP_Tag.php [hunk 1]
```diff
@@ -150,8 +143,6 @@ protected function render() {
 			$gtag_amp_opt_filtered['vars'] = $gtag_amp_opt['vars'];
 		}
 
-		$gtag_amp_opt_filtered['vars']['gtag_id'] = $this->tag_id;
```
Comment (aaemnnosttv) [inline includes/Modules/Analytics/AMP_Tag.php:153]: Note, this is still defined above in the definition of `$gtag_amp_opt`

**Review by aaemnnosttv — COMMENTED** [review]
A few extra notes for context

**Comment by aaemnnosttv** [comment]
Thanks @tofumatt! I've done some extra testing here and this seems good to go!

:shipit:

**Pull request merged**